includes/overlords/contribs.php

```diff
@@ -570,4 +570,167 @@ public static function build_sort()
+
+		if (!phpbb::$config['feed_overall'])
+		{
+			// Don't proceed if feeds are disabled
+			trigger_error('NO_FEED_ENABLED');
+		}
+
+		// Show one contribution only, if on the specific contrib page
+		$contrib_specific = ($contrib_id) ? 'AND r.contrib_id = ' . (int) $contrib_id : '';
+
+		$sql_ary = [
+			'SELECT' => 'r.*, c.*, u.username_clean',
+
+			'FROM' => [
+				TITANIA_REVISIONS_TABLE => 'r',
+				TITANIA_CONTRIBS_TABLE => 'c',
+				USERS_TABLE => 'u',
+			],
+
+			'WHERE'	=> 'r.revision_status = ' . ext::TITANIA_REVISION_APPROVED . '
+				AND r.revision_submitted = 1
+				AND c.contrib_status = ' . ext::TITANIA_CONTRIB_APPROVED . '
+				AND r.contrib_id = c.contrib_id
+				AND u.user_id = c.contrib_user_id 
+				' . $contrib_specific,
+
+			'ORDER_BY'	=> 'r.validation_date DESC',
+		];
+
+		$sql = phpbb::$db->sql_build_query('SELECT', $sql_ary);
+		$result = phpbb::$db->sql_query_limit($sql, 100);
+
+		$rows = [];
+		$feed_updated_time = false;
+
+		while ($row = phpbb::$db->sql_fetchrow($result))
+		{
+			// Only proceed if the contribution is not in categories that are all hidden
+			if (!self::feed_hidden_category_check($row['contrib_id']))
+			{
+				$feed_rows = [];
+				$feed_rows['item_date'] = date(\DateTime::ATOM, $row['validation_date']);
+
+				// Get the most recent time
+				if (!$feed_updated_time)
+				{
+					$feed_updated_time = $row['validation_date'];
+				}
+
+				// Make the name including the version
+				$feed_rows['item_title'] = $row['contrib_name'] . ' ' . $row['revision_version'];
+
+				if ($row['revision_name'])
+				{
+					// Include the code name if it's supplied
+					$feed_rows['item_title'] .= ' (' . $row['revision_name'] . ')';
+				}
+
+				$feed_rows['item_author'] = $row['username_clean'];
+				$feed_rows['item_description'] = phpbb::$user->lang('FEED_CDB_NEW_VERSION', $row['revision_version'], $row['contrib_name']);
+				$feed_rows['item_link'] = '';
+
+				if ($row['attachment_id'])
+				{
+					// Include the download link
+					$feed_rows['item_link'] = $helper->route('phpbb.titania.download', array('id' => $row['attachment_id']));
+				}
+
+				else
+				{
+					// This is so we can link to a bbCode customisation details page for example, because it doesn't
+					// have a download link.
+					if ($contrib)
+					{
+						$feed_rows['item_link'] = $contrib->get_url();
+					}
+
+					else
+					{
+						// Load the contribution if we don't have it already.
+						self::load_contrib($row['contrib_id']);
+						$feed_rows['item_link'] = self::get_contrib_object($row['contrib_id'])->get_url();
+					}
+				}
+
+				if ($feed_rows['item_link'])
+				{
+					// Strip session
+					$feed_rows['item_link'] = $path_helper->strip_url_params($feed_rows['item_link'], 'sid');
+				}
+
+				$rows[] = $feed_rows;
+			}
+		}
+
+		phpbb::$db->sql_freeresult($result);
+
+		if (!$rows)
+		{
+			// If there's no results, we can't proceed
+			trigger_error('FEED_CDB_NOT_AVAILABLE');
```

Comment:
If I understand it correctly, for example, when a contribution's only validated revision is pulled (e.g. for security reasons), is the feed supposed to throw an error? What's the expected behavior? Should we render a valid but empty feed here instead?

Reply:
That's correct, the feed is only served for contributions that have valid revisions.
But it's easy to change it to an empty feed if that behaviour would be preferred? @DavidIQ @VSEphpbb @kasimi

Reply:
I think that would be preferred, yes. No sense in triggering and likely logging an error if it's just someone following something they shouldn't.

Reply:
Done - I've changed it at e61c13fe90baf6821bbf65557e78f2a50d62cc09 so an empty feed will be displayed now.